resilience4j-feign/src/main/java/io/github/resilience4j/feign/PostponedDecorators.java

```diff
@@ -0,0 +1,139 @@
+        return this;
+    }
+
+    public PostponedDecorators<T> withRetry(Retry retryContext,
+        ScheduledExecutorService scheduler) {
+        operations.add(supplier ->
+            Retry.decorateCompletionStage(retryContext, scheduler, supplier));
+        return this;
+    }
+
+    public PostponedDecorators<T> withBulkhead(Bulkhead bulkhead) {
+        operations.add(supplier -> Bulkhead.decorateCompletionStage(bulkhead, supplier));
+        return this;
+    }
+
+    public PostponedDecorators<T> withTimeLimiter(TimeLimiter timeLimiter,
+        ScheduledExecutorService scheduler) {
+        operations.add(supplier -> timeLimiter.decorateCompletionStage(scheduler, supplier));
+        return this;
+    }
+
+    public PostponedDecorators<T> withRateLimiter(RateLimiter rateLimiter) {
+        return withRateLimiter(rateLimiter, 1);
+    }
+
+    public PostponedDecorators<T> withRateLimiter(RateLimiter rateLimiter, int permits) {
+        operations.add(supplier ->
+            RateLimiter.decorateCompletionStage(rateLimiter, permits, supplier));
+        return this;
+    }
+
+    public PostponedDecorators<T> withFallback(Predicate<T> resultPredicate,
+        UnaryOperator<T> resultHandler) {
+        operations.add(supplier ->
+            CompletionStageUtils.recover(supplier, resultPredicate, resultHandler));
+        return this;
+    }
+
+    public PostponedDecorators<T> withFallback(
+        BiFunction<T, Throwable, T> handler) {
+        operations.add(supplier -> CompletionStageUtils.andThen(supplier, handler));
+        return this;
+    }
+
+    public PostponedDecorators<T> withFallback(List<Class<? extends Throwable>> exceptionTypes,
+        Function<Throwable, T> exceptionHandler) {
+        operations.add(supplier ->
+            CompletionStageUtils.recover(supplier, exceptionTypes, exceptionHandler));
+        return this;
+    }
+
+    public PostponedDecorators<T> withFallback(Function<Throwable, T> exceptionHandler) {
+        operations.add(supplier -> CompletionStageUtils.recover(supplier, exceptionHandler));
+        return this;
+    }
+
+    public <X extends Throwable> PostponedDecorators<T> withFallback(Class<X> exceptionType,
+        Function<Throwable, T> exceptionHandler) {
+        operations.add(supplier ->
+            CompletionStageUtils.recover(supplier, exceptionType, exceptionHandler));
+        return this;
+    }
+
+    public PostponedDecorators<T> withFallbackFactory(Function<Exception, ?> fallbackFactory) {
+        throw new NotImplementedError("withFallbackFactory");
+//        decorators.add(new FallbackDecorator<>(new FallbackFactory<>(fallbackFactory)));
+//        return this;
+    }
+
+    public PostponedDecorators<T> withFallback(Object fallback) {
+        throw new NotImplementedError("withFallback Object");
+//        decorators.add(new FallbackDecorator<>(new DefaultFallbackHandler<>(fallback)));
```

Comment:
Here, a user wants to specify a class which implements the same interface as a fallback